components/zcash_address/src/kind/unified.rs

```diff
@@ -39,7 +39,37 @@ pub enum Typecode {
     Unknown(u32),
 }
 
-impl Typecode {
+impl TryFrom<u32> for DataTypecode {
+    type Error = ();
+
+    fn try_from(typecode: u32) -> Result<Self, Self::Error> {
+        match typecode {
+            0x00 => Ok(DataTypecode::P2pkh),
+            0x01 => Ok(DataTypecode::P2sh),
+            0x02 => Ok(DataTypecode::Sapling),
+            0x03 => Ok(DataTypecode::Orchard),
+            0x04..=0xDF | 0xFD..=0x02000000 => Ok(DataTypecode::Unknown(typecode)),
+            _ => Err(()),
```

Comment:
```suggestion
            // 0xC0..=0xDF are metadata typecodes that do not need to be understood.
            0x04..=0xDF | 0xFD..=0x02000000 => Ok(DataTypecode::Unknown(typecode)),
            // 0xE0..=0xFC are MUST-understand metadata typecodes. Since this library
            // does not understand them, they cause an error.
            _ => Err(()),
```